pandas/io/common.py

```diff
@@ -234,6 +236,8 @@ def get_filepath_or_buffer(
             ).open()
 
         return file_obj, encoding, compression, True
+    elif storage_options:
+        raise ValueError("storage_options passed with non-fsspec URL")
```

Comment:
Is the "URL" here actually a user-provided buffer?

Reply:
Local file or HTTP URL; but I guess an already open file instance would end up here too, worth checking.